gdc_rnaseq_tools/augment_star_counts.py

```diff
@@ -0,0 +1,228 @@
+        gene_name
+        gene_type
+        Chromosome
+
+    returns pandas DataFrame with all columns
+    '''
+    # load data
+    gdf = pd.read_table(gene_info)
+
+    # check that data has exactly the columns expected
+    expected_cols = [
+        'gene_id',
+        'total_exon_length',
+        'gene_name',
+        'gene_type',
+        'Chromosome',
+    ]
+    if set(expected_cols) - set(gdf.columns) != set():
+        raise DataFormatError('Expected columns not found')
+
+    # would be good to check that file is the correct length, but that's
+    # tied to the annotation and I'm not sure how to do that
+    return gdf
+
+
+def prep_data(counts, gene_info):
+    '''
+    Load data files and merge on gene_id via "inner" join operation.
+
+    STAR counts have 4 extra lines at the top reporting unmapped,
+    multimapping, noFeature, and ambiguous reads. These are not included
+    in the merged DataFrame, but are returned in the extras DataFrame to
+    be merged back into the final output.
+
+    Returns a tuple of:
+        merged DataFrame
+        extras DataFrame
+    '''
+    cdf = load_counts_table(counts)
+    gdf = load_gene_info(gene_info)
+    df = pd.merge(gdf, cdf, on='gene_id', how='inner')
+
+    if len(df) != len(gdf):
+        raise DataError(
+            'Data length or key mismatch between gene counts and gene '
+            + 'annotation tables. Possible file truncation.'
+        )
+
+    # save table of mis-aligned read stats also reported by star
+    mdf = cdf.iloc[0:4].copy()
+
+    return df, mdf
+
+
+def calc_tpm(df, excol='unstranded', lencol='total_exon_length'):
+    '''
+    Transcripts Per Million
+
+    Calculate the TPM values for all genes in DataFrame df
+
+    TPM = RPK x 1e6 / M
+
+        RPK = C X 1e3 / L
+        C : count of fragments aligned to this gene
+        L : sum of exon lengths in gene where overlapping exons are merged
+            otherwise known as union exon length
+        M : sum over all genes of RPK values
+    '''
+    # RPK - reads per thousand bp of transcript length
+    rpk = df[excol] * 1e3 / df[lencol]
+    # sum of RPK signal
+    M = rpk.sum()
+    # TPM - transcripts per million
+    tpm = rpk * 1e6 / M
+    return tpm
+
+
+def calc_fpkm(df, excol='unstranded', lencol='total_exon_length', typecol='gene_type'):
+    '''
+    Fragments Per Kilobases (of transcript) and Millions (of fragments)
+
+    Calculate the FPKM values for all genes in DataFrame df
+
+    FPKM = (C × 1e9)/(NL)
+
+        C : count of fragments aligned to this gene
+        N : total fragment count to protein-coding genes
+        L : sum of exon lengths in gene where overlapping exons are merged
+    '''
+    # select protein coding genes
+    sel = df[typecol] == 'protein_coding'
+    # get sum of counts in protein coding genes
+    N = df.loc[sel][excol].sum()
+    # calculate fpkm
+    fpkm = df[excol] * 1e9 / (N * df[lencol])
+
+    return fpkm
+
+
+def calc_fpkm_uq(
+    df,
```

Comment:
For important strings like these, I'd recommend using an enum

```python
from enum import Enum

class ColNames(Enum):
    UNSTRANDED = "unstranded"
    LENGTH = "total_exon_length"
    ...
```
And in the function you would access the values like `ColNames.LENGTH.value`

You could also create enums for the non-numeric chromosomes and gene types (`"protein_coding"`, etc)